### ISSUE TITLE
IamRoleArn missing for S3 action

### ISSUE THREAD

**Issue by josefguenther** [aws-cloudformation/cfn-lint#4188]
### CloudFormation Lint Version

cfn-lint 1.37.2

### What operating system are you using?

Mac

### Describe the bug

AWS::SES::ReceiptRule
-> Rule
-> Actions
-> S3Action

I get: Property IamRoleArn is not allowed.yaml-schema: S3Action(513)

https://docs.aws.amazon.com/AWSCloudFormation/latest/TemplateReference/aws-properties-ses-receiptrule-s3action.html

### Expected behavior

This is a valid property in the schema and deploys without issue.

### Reproduction template

```yaml
  IncomingEmailRule:
    Type: AWS::SES::ReceiptRule
    Properties:
      Rule:
        Actions:
          - S3Action:
              BucketName: !Ref MainBucket
              IamRoleArn: !GetAtt IncomingEmailRole.Arn
              ObjectKeyPrefix: inbox/email/
              TopicArn: !Ref IncomingEmailTopic
```
